### ISSUE TITLE
ScriptThread stores mousemove target nodes and invokes content JS on closed pipelines

### ISSUE THREAD

**Issue by aguai** [servo/servo#16057]
called `Result::unwrap()` on an `Err` value: "SendError(..)"
### Servo Version:
Servo 0.0.1-4cd7321
### Backtrace:
```

WARNING: <Constellation>: Sending layout exit message failed (Bogus destination port.).
WARNING: <Constellation>: creating replacement pipeline for about:failure
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(12) }>: Panic hook called.
ERROR: called `Result::unwrap()` on an `Err` value: "SendError(..)"

stack backtrace:
   0:        0x10ffc160e - backtrace::backtrace::trace::h855657dbc57f873e
   1:        0x10ffc191c - backtrace::capture::Backtrace::new::h023332005d95d7ee
   2:        0x10ffb97b5 - constellation::constellation::log_entry::hf40e10dc2983df6f
   3:        0x10ffb8692 - <constellation::constellation::FromCompositorLogger as log::Log>::log::h1afed46b510e4269
   4:        0x111a1bb61 - log::__log::h4694da532ab3af11
   5:        0x10ff149c0 - servo::main::{{closure}}::h19fe47667a350ff0
   6:        0x111a964d3 - std::panicking::rust_panic_with_hook::h3e771863e3ed8403
   7:        0x111a96354 - std::panicking::begin_panic::h0af46715c5e7593e
   8:        0x111a962c2 - std::panicking::begin_panic_fmt::h9509a584fa230ae8
   9:        0x111a96227 - rust_begin_unwind
  10:        0x111abf8c0 - core::panicking::panic_fmt::h5c28b8150c75bd71
  11:        0x1107cd900 - core::result::unwrap_failed::h68569651dedff04f
  12:        0x110ac739f - script::dom::window::Window::force_reflow::hcc18b271c23d6a0c
  13:        0x110ac7975 - script::dom::window::Window::reflow::he71857c1754cfdac
  14:        0x110b0dd7d - script::script_thread::ScriptThread::handle_msgs::h4b43f852f97569f7
  15:        0x1107849a7 - std::panicking::try::do_call::hfb5a6d3eb1b24f47
  16:        0x111a973aa - __rust_maybe_catch_panic
  17:        0x1107f7816 - <F as alloc::boxed::FnBox<A>>::call_box::h749677ac1c228423
  18:        0x111a95454 - std::sys::imp::thread::Thread::new::thread_start::h5f22426d80277e16
  19:     0x7fff930b699c - _pthread_body
  20:     0x7fff930b6919 - _pthread_start
```
This report was generated by the browser.html issue reporter.

**Comment by aguai** [comment]
error

**Comment by jdm** [comment]
@aguai What page were you looking at?

**Comment by odeits** [comment]
Got something similar when trying to reload a page (bitbucket)

```
called `Result::unwrap()` on an `Err` value: "SendError(..)" (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }, at src/libcore/result.rs:859)
stack backtrace:
   0:        0x10d1dfabe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x10d1e020c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x10cbce218 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::hc2ae472b5acad1df
   3:        0x10e780b99 - std::panicking::rust_panic_with_hook::h05996066754c6be9
   4:        0x10e780a44 - std::panicking::begin_panic::h9fecf34da42eb910
   5:        0x10e780962 - std::panicking::begin_panic_fmt::he5aad713258a67c3
   6:        0x10e7808c7 - rust_begin_unwind
   7:        0x10e7a8cd0 - core::panicking::panic_fmt::he26d734b771c5b2c
   8:        0x10d424510 - core::result::unwrap_failed::hdaafa0cb0aa0b8e6
   9:        0x10d6b2e47 - script::dom::window::Window::force_reflow::h5edb8958fd8594f4
  10:        0x10d6b4ffe - script::dom::window::Window::reflow::h57c48d129edf3155
  11:        0x10d545249 - script::dom::cssstyledeclaration::CSSStyleDeclaration::get_property_value::hfda8e497678ec11b
  12:        0x10d6f6825 - script::script_thread::ScriptThread::handle_msg_from_constellation::h3ead274e9334db51
  13:        0x10d6f4f35 - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::hdb847b2d7ecb3d5d
  14:        0x10d6f17f1 - script::script_thread::ScriptThread::handle_msgs::h20b0bbbd8da7bef9
  15:        0x10d276617 - std::panicking::try::do_call::h0c991146010a3143
  16:        0x10e781b6a - __rust_maybe_catch_panic
  17:        0x10d425986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  18:        0x10e77d6b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  19:     0x7fff87b1bc12 - _pthread_body
  20:     0x7fff87b1bb8f - _pthread_start
ERROR:servo: called `Result::unwrap()` on an `Err` value: "SendError(..)"
Stack trace for thread "ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }"
stack backtrace:
   0:        0x10d1dfabe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x10d1e020c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x10cbcb83c - servo::install_crash_handler::handler::h56272dfc22f55ee8
   3:     0x7fff93cfcea9 - _sigtramp
   4:        0x10e007617 - void DispatchToTracer<JSObject*>(JSTracer*, JSObject**, char const*)
   5:        0x10d4eeab2 - script::dom::bindings::refcounted::trace_refcounted_objects::h97369e4d336639bd
   6:        0x10e0014f1 - js::gc::GCRuntime::markRuntime(JSTracer*, js::gc::GCRuntime::TraceOrMarkRuntime, js::AutoLockForExclusiveAccess&)
   7:        0x10ddcdc4a - js::gc::GCRuntime::beginMarkPhase(JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
   8:        0x10ddd3738 - js::gc::GCRuntime::incrementalCollectSlice(js::SliceBudget&, JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
   9:        0x10ddd4078 - js::gc::GCRuntime::gcCycle(bool, js::SliceBudget&, JS::gcreason::Reason)
  10:        0x10ddd45a3 - js::gc::GCRuntime::collect(bool, js::SliceBudget, JS::gcreason::Reason)
  11:        0x10ddd4888 - js::gc::GCRuntime::gc(JSGCInvocationKind, JS::gcreason::Reason)
  12:        0x10ded9d77 - JSRuntime::destroyRuntime()
  13:        0x10dd916b5 - js::DestroyContext(JSContext*)
  14:        0x10da0d040 - _$LT$js..rust..Runtime$u20$as$u20$core..ops..Drop$GT$::drop::h190782c749a12e00
  15:        0x10d3f464d - core::ptr::drop_in_place::h37533037635b4dd7
  16:        0x10d276c8c - std::panicking::try::do_call::h0c991146010a3143
  17:        0x10e781b6a - __rust_maybe_catch_panic
  18:        0x10d425986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  19:        0x10e77d6b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  20:     0x7fff87b1bc12 - _pthread_body
  21:     0x7fff87b1bb8f - _pthread_start
Stack trace for thread "ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }"
stack backtrace:
   0:        0x10d1dfabe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x10d1e020c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x10cbcb83c - servo::install_crash_handler::handler::h56272dfc22f55ee8
   3:     0x7fff93cfcea9 - _sigtramp
   4:        0x10cbcb8c0 - servo::install_crash_handler::handler::h56272dfc22f55ee8
   5:     0x7fff93cfcea9 - _sigtramp
   6:        0x10e007617 - void DispatchToTracer<JSObject*>(JSTracer*, JSObject**, char const*)
   7:        0x10d4eeab2 - script::dom::bindings::refcounted::trace_refcounted_objects::h97369e4d336639bd
   8:        0x10e0014f1 - js::gc::GCRuntime::markRuntime(JSTracer*, js::gc::GCRuntime::TraceOrMarkRuntime, js::AutoLockForExclusiveAccess&)
   9:        0x10ddcdc4a - js::gc::GCRuntime::beginMarkPhase(JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
  10:        0x10ddd3738 - js::gc::GCRuntime::incrementalCollectSlice(js::SliceBudget&, JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
  11:        0x10ddd4078 - js::gc::GCRuntime::gcCycle(bool, js::SliceBudget&, JS::gcreason::Reason)
  12:        0x10ddd45a3 - js::gc::GCRuntime::collect(bool, js::SliceBudget, JS::gcreason::Reason)
  13:        0x10ddd4888 - js::gc::GCRuntime::gc(JSGCInvocationKind, JS::gcreason::Reason)
  14:        0x10ded9d77 - JSRuntime::destroyRuntime()
  15:        0x10dd916b5 - js::DestroyContext(JSContext*)
  16:        0x10da0d040 - _$LT$js..rust..Runtime$u20$as$u20$core..ops..Drop$GT$::drop::h190782c749a12e00
  17:        0x10d3f464d - core::ptr::drop_in_place::h37533037635b4dd7
  18:        0x10d276c8c - std::panicking::try::do_call::h0c991146010a3143
  19:        0x10e781b6a - __rust_maybe_catch_panic
  20:        0x10d425986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  21:        0x10e77d6b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  22:     0x7fff87b1bc12 - _pthread_body
  23:     0x7fff87b1bb8f - _pthread_start
```

**Comment by jdm** [comment]
Is there a specific bitbucket page?

**Comment by odeits** [comment]
@jdm this is `Atlassian Bitbucket v4.11.2` anything after the log in page. The display says "Could not load the requested page: Unexpected scheme" I am on OSX and I hit command + r

This is running ./mach run --release https://bitbucket.url.here

full output below

```
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:857:396
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:863:344
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2278:166
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2348:148
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2396:350
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2458:447
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:857:396
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:863:344
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2278:166
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2348:148
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2396:350
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2458:447
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:857:396
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:863:344
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2278:166
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2348:148
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2396:350
DEPRECATED JS - Cookie has been deprecated since 5.8.0 and will be removed in a future release. Use cookie instead.

 @https://bitbucket.vidder.net/s/816e401fec0f8e71ce8adbe58a33ccc9-CDN/1354557670/dfbc6ae/1/ae1be24ea8754179495e60ec2eaba772/_/download/contextbatch/js/_super/batch.js?locale=en-US:2458:447
Deprecation warning: moment().zone is deprecated, use moment().utcOffset instead. https://github.com/moment/moment/issues/1779
Deprecation warning: moment().add(period, number) is deprecated. Please use moment().add(number, period).
Deprecation warning: moment().subtract(period, number) is deprecated. Please use moment().subtract(number, period).
DEPRECATED JS - Inline dialog constructor has been deprecated and will be removed in a future release. Use inline dialog 2 instead.

 @https://bitbucket.vidder.net/s/fab8651e979e075ca72f77b16fe8b8fe-T/1354557670/dfbc6ae/1/3041c335f82b20549c051057c9d257be/_/download/contextbatch/js/bitbucket.page.dashboard,bitbucket.layout.base,atl.general,-_super/batch.js?locale=en-US&nps-not-opted-out=true&supportedApplication=true:2132:42
called `Result::unwrap()` on an `Err` value: "SendError(..)" (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }, at src/libcore/result.rs:859)
stack backtrace:
   0:        0x101f87abe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x101f8820c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x101976218 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::hc2ae472b5acad1df
   3:        0x103528b99 - std::panicking::rust_panic_with_hook::h05996066754c6be9
   4:        0x103528a44 - std::panicking::begin_panic::h9fecf34da42eb910
   5:        0x103528962 - std::panicking::begin_panic_fmt::he5aad713258a67c3
   6:        0x1035288c7 - rust_begin_unwind
   7:        0x103550cd0 - core::panicking::panic_fmt::he26d734b771c5b2c
   8:        0x1021cc510 - core::result::unwrap_failed::hdaafa0cb0aa0b8e6
   9:        0x10245ae47 - script::dom::window::Window::force_reflow::h5edb8958fd8594f4
  10:        0x10245cffe - script::dom::window::Window::reflow::h57c48d129edf3155
  11:        0x1022ed249 - script::dom::cssstyledeclaration::CSSStyleDeclaration::get_property_value::hfda8e497678ec11b
  12:        0x10249e825 - script::script_thread::ScriptThread::handle_msg_from_constellation::h3ead274e9334db51
  13:        0x10249cf35 - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::hdb847b2d7ecb3d5d
  14:        0x1024997f1 - script::script_thread::ScriptThread::handle_msgs::h20b0bbbd8da7bef9
  15:        0x10201e617 - std::panicking::try::do_call::h0c991146010a3143
  16:        0x103529b6a - __rust_maybe_catch_panic
  17:        0x1021cd986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  18:        0x1035256b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  19:     0x7fff87b1bc12 - _pthread_body
  20:     0x7fff87b1bb8f - _pthread_start
ERROR:servo: called `Result::unwrap()` on an `Err` value: "SendError(..)"
Stack trace for thread "ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }"
stack backtrace:
   0:        0x101f87abe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x101f8820c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x10197383c - servo::install_crash_handler::handler::h56272dfc22f55ee8
   3:     0x7fff93cfcea9 - _sigtramp
   4:        0x102daf617 - void DispatchToTracer<JSObject*>(JSTracer*, JSObject**, char const*)
   5:        0x102296ab2 - script::dom::bindings::refcounted::trace_refcounted_objects::h97369e4d336639bd
   6:        0x102da94f1 - js::gc::GCRuntime::markRuntime(JSTracer*, js::gc::GCRuntime::TraceOrMarkRuntime, js::AutoLockForExclusiveAccess&)
   7:        0x102b75c4a - js::gc::GCRuntime::beginMarkPhase(JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
   8:        0x102b7b738 - js::gc::GCRuntime::incrementalCollectSlice(js::SliceBudget&, JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
   9:        0x102b7c078 - js::gc::GCRuntime::gcCycle(bool, js::SliceBudget&, JS::gcreason::Reason)
  10:        0x102b7c5a3 - js::gc::GCRuntime::collect(bool, js::SliceBudget, JS::gcreason::Reason)
  11:        0x102b7c888 - js::gc::GCRuntime::gc(JSGCInvocationKind, JS::gcreason::Reason)
  12:        0x102c81d77 - JSRuntime::destroyRuntime()
  13:        0x102b396b5 - js::DestroyContext(JSContext*)
  14:        0x1027b5040 - _$LT$js..rust..Runtime$u20$as$u20$core..ops..Drop$GT$::drop::h190782c749a12e00
  15:        0x10219c64d - core::ptr::drop_in_place::h37533037635b4dd7
  16:        0x10201ec8c - std::panicking::try::do_call::h0c991146010a3143
  17:        0x103529b6a - __rust_maybe_catch_panic
  18:        0x1021cd986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  19:        0x1035256b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  20:     0x7fff87b1bc12 - _pthread_body
  21:     0x7fff87b1bb8f - _pthread_start
Stack trace for thread "ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }"
stack backtrace:
   0:        0x101f87abe - backtrace::backtrace::trace::h47a32bf8c6f00c9b
   1:        0x101f8820c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:        0x10197383c - servo::install_crash_handler::handler::h56272dfc22f55ee8
   3:     0x7fff93cfcea9 - _sigtramp
   4:        0x1019738c0 - servo::install_crash_handler::handler::h56272dfc22f55ee8
   5:     0x7fff93cfcea9 - _sigtramp
   6:        0x102daf617 - void DispatchToTracer<JSObject*>(JSTracer*, JSObject**, char const*)
   7:        0x102296ab2 - script::dom::bindings::refcounted::trace_refcounted_objects::h97369e4d336639bd
   8:        0x102da94f1 - js::gc::GCRuntime::markRuntime(JSTracer*, js::gc::GCRuntime::TraceOrMarkRuntime, js::AutoLockForExclusiveAccess&)
   9:        0x102b75c4a - js::gc::GCRuntime::beginMarkPhase(JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
  10:        0x102b7b738 - js::gc::GCRuntime::incrementalCollectSlice(js::SliceBudget&, JS::gcreason::Reason, js::AutoLockForExclusiveAccess&)
  11:        0x102b7c078 - js::gc::GCRuntime::gcCycle(bool, js::SliceBudget&, JS::gcreason::Reason)
  12:        0x102b7c5a3 - js::gc::GCRuntime::collect(bool, js::SliceBudget, JS::gcreason::Reason)
  13:        0x102b7c888 - js::gc::GCRuntime::gc(JSGCInvocationKind, JS::gcreason::Reason)
  14:        0x102c81d77 - JSRuntime::destroyRuntime()
  15:        0x102b396b5 - js::DestroyContext(JSContext*)
  16:        0x1027b5040 - _$LT$js..rust..Runtime$u20$as$u20$core..ops..Drop$GT$::drop::h190782c749a12e00
  17:        0x10219c64d - core::ptr::drop_in_place::h37533037635b4dd7
  18:        0x10201ec8c - std::panicking::try::do_call::h0c991146010a3143
  19:        0x103529b6a - __rust_maybe_catch_panic
  20:        0x1021cd986 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h4859557d9c429816
  21:        0x1035256b4 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  22:     0x7fff87b1bc12 - _pthread_body
  23:     0x7fff87b1bb8f - _pthread_start
```

**Comment by jdm** [comment]
I haven't been able to reproduce this when visiting https://bitbucket.org/PrimordialHelios/, logging in, and then clicking around on the logged in page.

**Comment by jdm** [comment]
Ok, I can reproduce both results by refreshing the page while it's loading:
```
   8:        0x10181bb10 - core::result::unwrap_failed<std::sync::mpsc::SendError<script_layout_interface::message::Msg>>
                        at /Users/travis/build/rust-lang/rust/src/libcore/macros.rs:29
   9:        0x10216fcba - core::result::{{impl}}::unwrap<(),std::sync::mpsc::SendError<script_layout_interface::message::Msg>>
                        at /Users/travis/build/rust-lang/rust/src/libcore/result.rs:737
  10:        0x10254120e - script::dom::window::Window::force_reflow::hfb0c8f1e95e4d56a
                        at /Users/jdm/src/master-servo/components/script/dom/window.rs:1242
  11:        0x10254283b - script::dom::window::Window::reflow::h0d859229285eb440
                        at /Users/jdm/src/master-servo/components/script/dom/window.rs:1321
  12:        0x102543512 - script::dom::window::Window::client_rect_query::h2f209e43d21e0a67
                        at /Users/jdm/src/master-servo/components/script/dom/window.rs:1392
  13:        0x101a3c437 - script::dom::node::Node::client_rect::h55d1c83eab492436
                        at /Users/jdm/src/master-servo/components/script/dom/node.rs:580
  14:        0x100c63c8d - _$LT$script..dom..element..Element$u20$as$u20$script..dom..bindings..codegen..Bindings..ElementBinding..ElementBinding..ElementMethods$GT$::ClientLeft::h93d4f96ea2ef1a51
                        at /Users/jdm/src/master-servo/components/script/dom/element.rs:1912
  15:        0x101325732 - script::dom::bindings::codegen::Bindings::ElementBinding::ElementBinding::get_clientLeft::{{closure}}
                        at /Users/jdm/src/master-servo/target/debug/build/script-e76373ee7ebff941/out/Bindings/ElementBinding.rs:2762
```
as well as
```
   9:        0x101f114f2 - core::option::{{impl}}::unwrap<script::dom::bindings::js::Root<script::dom::windowproxy::WindowProxy>>
                        at /Users/travis/build/rust-lang/rust/src/libcore/macros.rs:21
  10:        0x102539782 - script::dom::window::Window::window_proxy::hc882545fad82990a
                        at /Users/jdm/src/master-servo/components/script/dom/window.rs:342
  11:        0x10253bd33 - _$LT$script..dom..window..Window$u20$as$u20$script..dom..bindings..codegen..Bindings..WindowBinding..WindowBinding..WindowMethods$GT$::Window::h7659a069ed0f399d
                        at /Users/jdm/src/master-servo/components/script/dom/window.rs:646
  12:        0x101549869 - script::dom::bindings::codegen::Bindings::WindowBinding::WindowBinding::get_window::{{closure}}
                        at /Users/jdm/src/master-servo/target/debug/build/script-e76373ee7ebff941/out/Bindings/WindowBinding.rs:779
  13:        0x100ffc4c7 - core::ops::FnOnce::call_once<closure,()>
                        at /Users/travis/build/rust-lang/rust/src/libcore/ops.rs:2683
  14:        0x10115b76a - std::panic::{{impl}}::call_once<bool,closure>
                        at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:296
  15:        0x1009deeb7 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,bool>
                        at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:454
  16:        0x105733bda - __rust_maybe_catch_panic
                        at src/libpanic_unwind/lib.rs:98
  17:        0x1008f979d - std::panicking::try<bool,std::panic::AssertUnwindSafe<closure>>
                        at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:433
  18:        0x100f51301 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,bool>
                        at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:361
  19:        0x100de0960 - js::panic::wrap_panic<std::panic::AssertUnwindSafe<closure>,bool>
                        at /Users/jdm/.cargo/git/checkouts/rust-mozjs-8611526964119dd6/08af365/src/panic.rs:22
  20:        0x101bbe5ec - script::dom::bindings::codegen::Bindings::WindowBinding::WindowBinding::get_window
                        at /Users/jdm/src/master-servo/target/debug/build/script-e76373ee7ebff941/out/Bindings/WindowBinding.rs:777
  21:        0x1030b9ada - CallJitGetterOp
                        at /Users/jdm/.cargo/git/checkouts/rust-mozjs-8611526964119dd6/08af365/src/jsglue.cpp:498
  22:        0x1019d7fd1 - script::dom::bindings::utils::generic_call
                        at /Users/jdm/src/master-servo/components/script/dom/bindings/utils.rs:446
  23:        0x1019d8141 - script::dom::bindings::utils::generic_getter
                        at /Users/jdm/src/master-servo/components/script/dom/bindings/utils.rs:462
  24:        0x10352dd74 - js::InternalCallOrConstruct(JSContext*, JS::CallArgs const&, js::MaybeConstruct)
                        at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/jscntxtinlines.h:232
  25:        0x10355121b - CallGetter(JSContext*, JS::Handle<JSObject*>, JS::Handle<JS::Value>, JS::Handle<js::Shape*>, JS::MutableHandle<JS::Value>)
                        at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:631
  26:        0x103544b74 - js::NativeGetExistingProperty(JSContext*, JS::Handle<JSObject*>, JS::Handle<js::NativeObject*>, JS::Handle<js::Shape*>, JS::MutableHandle<JS::Value>)
                        at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/NativeObject.cpp:1785
  27:        0x10354c4d4 - bool js::FetchName<false>(JSContext*, JS::Handle<JSObject*>, JS::Handle<JSObject*>, JS::Handle<js::PropertyName*>, JS::Handle<js::Shape*>, JS::MutableHandle<JS::Value>)
                        at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter-inl.h:202
  28:        0x1035288ff - Interpret(JSContext*, js::RunState&)
                        at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:230
```

**Comment by jdm** [comment]
The panic under ClientLeft happens because we're running JS in a window that has had `clear_js_runtime` called.

**Comment by jdm** [comment]
It's very strange - we correctly receive a mouse event targeted at the new pipeline that was created as a result of the refresh. This dispatches a `move` DOM event at the page, which causes a bunch of content JS to be invoked. By the time it ends up invoking `Element::ClientLeft`, when we call `Node::client_rect()` the node document is associated with the pre-reload window which is for a pipeline that has already been closed. This is very peculiar.

**Comment by jdm** [comment]
Aha! It's caused by the script thread storing the previous node that was the target of a mousemove event and dispatching a `mouseout` at it. We should not be mixing elements between documents; that's pretty bad!

**Comment by jdm** [comment]
Here is a minimal testcase showing this problem:
```html
<div onmouseover="location.reload()" onmouseout="console.log(this.clientLeft)">hi there</div>
```

**Comment by jdm** [comment]
One potential solution: in `ScriptThread::handle_exit_pipeline_msg`, we compare the document for `self.topmost_mouse_over_target` to the document obtained from `self.documents`. If they are the same, then we null out `self.topmost_mouse_over_target`.

**Comment by ibmandura** [comment]
I would like to take a shot at this one, but I am unable to reproduce the bug with the minimal testcase given here. `./mach run --release minimal.html`, where `minimal.html` consists of `<div onmouseover="location.reload()" onmouseout="this.clientLeft">hi there</div>`, does not seem to run into any problems. Can somebody confirm or deny this?

**Comment by jdm** [comment]
@ibmandura Did you move the mouse over the text?

**Comment by ibmandura** [comment]
@jdm yes I did, nothing happened

**Comment by jdm** [comment]
Huh. Something that merged in the past week has hidden the problem that my testcase was previously exposing.

**Comment by jdm** [comment]
Nevermind; it can only be reproduced when run under an HTTP server instead of directly from a file:// URL. I just got it to work using `python -m SimpleHTTPServer` in the same directory.

**Comment by atouchet** [comment]
Should this be marked as fixed now with #17323 merged?

**Comment by jdm** [comment]
Yep!
